src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/network/v2/AwsV2NetworkPlugin.java

```diff
@@ -1,46 +1,299 @@
+				.cidrBlock(cidr)
+				.vpcId(this.vpcId)
+				.build();
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		String subnetId = doCreateSubnetResquests(name, request, client);
+		doCreateSecurityGroupRequests(cidr, subnetId, client);
+		return subnetId;
+	}
+
+	@Override
+	public NetworkInstance getInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.GETTING_INSTANCE, networkOrder.getInstanceId(), cloudUser.getToken()));
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		String subnetId = networkOrder.getInstanceId();
+		Subnet subnet = getSubnetById(subnetId, client);
+		RouteTable routeTable = getRouteTables(client);
+		return mountNetworkInstance(subnet, routeTable);
+	}
+
+	@Override
+	public void deleteInstance(NetworkOrder networkOrder, AwsV2User cloudUser) throws FogbowException {
+		LOGGER.info(String.format(Messages.Info.DELETING_INSTANCE, networkOrder.getInstanceId(), cloudUser.getToken()));
+
+		Ec2Client client = AwsV2ClientUtil.createEc2Client(cloudUser.getToken(), this.region);
+		String subnetId = networkOrder.getInstanceId();
+		doDeleteSubnetRequests(subnetId, client);
+	}
+
+	@Override
+	public boolean isReady(String instanceState) {
+		return AwsV2StateMapper.map(ResourceType.NETWORK, instanceState).equals(InstanceState.READY);
+	}
+
+	@Override
+	public boolean hasFailed(String instanceState) {
+		return false;
+	}
+	
+	protected String getGroupIdBySubnet(String subnetId, Ec2Client client)
+			throws UnexpectedException, InstanceNotFoundException {
+		
+		Subnet subnet = getSubnetById(subnetId, client);
+		for (Tag tag : subnet.tags()) {
+			if (tag.key().equals(AWS_TAG_GROUP_ID)) {
+				return tag.value();
+			}
+		}
+		throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+	}
+
+	private NetworkInstance mountNetworkInstance(Subnet subnet, RouteTable routeTable) throws UnexpectedException {
+		String id = subnet.subnetId();
+		String cloudState = subnet.stateAsString();
+		String name = subnet.tags().listIterator().next().value();
+		String cidr = subnet.cidrBlock();
+		String gateway = getGatewayFromRouteTables(routeTable.routes());
+		String vLAN = null;
+		String networkInterface = null;
+		String macInterface = null;
+		String interfaceState = null;
+		NetworkAllocationMode networkAllocationMode = NetworkAllocationMode.DYNAMIC;
+		return new NetworkInstance(id, cloudState, name, cidr, gateway, vLAN, networkAllocationMode, networkInterface,
+				macInterface, interfaceState);
+	}
+
+	private String getGatewayFromRouteTables(List<Route> routes) {
+		for (Route route : routes) {
+			if (!route.gatewayId().equals(LOCAL_GATEWAY_DESTINATION)) {
+				return route.destinationCidrBlock();
+			}
+		}
+		return null;
+	}
+
+	private RouteTable getRouteTables(Ec2Client client) throws UnexpectedException, InstanceNotFoundException {
```

Comment:
throws FogbowException might be enough

Reply:
Consider them made, I'll fix it tomorrow.